rcl/test/rcl/test_arguments.cpp

```diff
@@ -194,6 +167,56 @@ TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), check_known_vs_unkno
   EXPECT_FALSE(are_known_ros_args({"--ros-args", "__log_level:=foo"}));
 }
 
+bool
+are_valid_ros_args(std::vector<const char *> argv)
+{
+  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
+  rcl_ret_t ret = rcl_parse_arguments(
+    argv.size(), argv.data(), rcl_get_default_allocator(), &parsed_args);
+  if (RCL_RET_OK != ret) {
+    rcl_reset_error();
+    return false;
+  }
+  EXPECT_EQ(RCL_RET_OK, rcl_arguments_fini(&parsed_args));
+  return true;
+}
+
+TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), check_valid_vs_invalid_args) {
+  EXPECT_TRUE(are_valid_ros_args({"--ros-args", "-p", "foo:=bar", "-r", "__node:=node_name"}));
+
+  EXPECT_FALSE(are_valid_ros_args({"--ros-args", "-r"}));
```

Comment:
Instead of checking for any error, I would explicitly check for `RCL_RET_INVALID_ROS_ARGS`.

Reply:
That makes sense.